actionpack/lib/action_dispatch.rb

```diff
@@ -66,8 +66,10 @@ class IllegalStateError < StandardError
     autoload :ShowExceptions
     autoload :SSL
     autoload :Static
+    autoload :ServerTiming
   end
 
+  autoload :Current
```

Comment:
I was a bit conflicted with this approach, of using `ActionDispatch::Current < ActiveSupport::CurrentAttributes`. 

I found the original `Thread.current` implementation much simpler or perhaps even using a `thread_mattr_accessor`. 

But what I like about this, is that `CurrentAttributes` takes care of resetting the Thread locals after a request cycle, I'm not sure we have that with `Thread.current` or `thread_mattr_accessor`.

Reply:
We don't need to use current attributes. We have control over the code so we can also take care of resetting the thread local we want to use after the request cycle.

Reply:
Ah, we are also using the thread local is a middleware so that same middleware can take care of cleaning itself.

Reply:
I just realized this as well, I'll update.